### DISCUSSION TITLE
In bun compile how can I embed everything inside a directory

### DISCUSSION THREAD

**Discussion by kivS** [oven-sh/bun#6653] [Q&A]
I want to embed a nextJS static export into an executable program, like Golang does with embed, eg `//go:embed _ui/out/*`

How can I accomplish this Bun? I know you can load a file and make it available internally :

```
// this becomes an internal file path
import icon from "./icon.png";

import { file } from "bun";

export default {
  fetch(req) {
    return new Response(file(icon));
  },
};
```

Is there a way to do the same for, say, `dist/*`?

**Comment by Exitium-DEV** [comment]
+1
did you find an answer?

**Reply by kivS** [reply]
For the moment I went with this workaround: https://github.com/seveibar/make-vfs/tree/main

**Reply by Exitium-DEV** [reply]
https://bun.sh/blog/bun-v1.1.25#embedding-files-in-standalone-executables-without-importing

They literally released an update last night that allows us to do it!

**Reply by UltiRequiem** [reply]
Thanks!

**Comment by UltiRequiem** [comment]
Please checkout https://bun.sh/blog/bun-v1.1.25#embedding-files-in-standalone-executables-without-importing
<img width="716" alt="image" src="https://github.com/user-attachments/assets/b17e98bb-ee98-43f7-b677-579aa13fdbe3" />

Mark as solved if this works for you.

**Comment by NobleMajo** [comment]
linux cachyos bash, tried to embed a dir:
```bash
test-project/client via 🥟 v1.3.1 
❯ bun build --compile ./src/index.ts ./embed/**/*.png
error: ModuleNotFound resolving "./embed/**/*.png" (entry point)

test-project/client via 🥟 v1.3.1 
❯ bun build --compile ./src/index.ts ./embed/**/*
error: ModuleNotFound resolving "./embed/**/*" (entry point)

test-project/client via 🥟 v1.3.1 
❯ bun -v
1.3.1

test-project/client via 🥟 v1.3.1 
❯ ll src/index.ts 
-rw-r--r-- 1 majo majo 440 29. Okt 20:18 src/index.ts

test-project/client via 🥟 v1.3.1 
❯ ll embed/
total 588
drwxr-xr-x 1 majo majo     72 29. Okt 19:35 .
drwxr-xr-x 1 majo majo    180 29. Okt 20:03 ..
-rw-r--r-- 1 majo majo   1657 29. Okt 19:27 index.html
-rw-r--r-- 1 majo majo    833 29. Okt 19:27 index.ts
-rw-r--r-- 1 majo majo 587739 26. Okt 04:06 logo.png
-rw-r--r-- 1 majo majo   2626 29. Okt 19:26 styles.css
```

id does build but give me an error when selecting the embed dir without start:
```bash
desktop/maudio/client via 🥟 v1.3.1 
❯ bun build --compile ./src/index.ts ./embed
   [3ms]  bundle  8 modules
   [5ms] compile  index
   
desktop/maudio/client via 🥟 v1.3.1 
❯ ./index 
EACCES: permission denied, mkdir '/$bunfs'
    path: "/$bunfs",
 syscall: "mkdir",
   errno: -13,
    code: "EACCES"


Bun v1.3.1 (Linux x64)
```

please fix docs and/or implementation +1

**Comment by NobleMajo** [comment]
I found a solution that worked for me — maybe it will work for you too.
Please consider updating the documentation.

The following works for me now, but it is not documented:
```bash
desktop/maudio/client via 🥟 v1.3.1 
❯ bun build --compile ./src/index.ts ./embed/**
   [3ms]  bundle  13 modules
  [17ms] compile  index
```

The following will break:
```bash
desktop/maudio/client via 🥟 v1.3.1 took 24s 
❯ bun build --compile ./src/index.ts ./embed/**/*
error: ModuleNotFound resolving "./embed/**/*" (entry point)
```

The following will not import the files from the './embed' directory (see `modules` count):
```bash
desktop/maudio/client via 🥟 v1.3.1 
❯ bun build --compile ./src/index.ts ./embed
   [3ms]  bundle  8 modules
   [3ms] compile  index
```

This behaviour is very strange and should be documented.

**Reply by ajaykarthikr** [reply]
How does this work during development, doesn't look like `embeddedFiles` works on dev.
